src/ucp/core/ucp_mm.c

```diff
@@ -1265,25 +1265,34 @@ void ucp_mem_rcache_cleanup(ucp_context_h context)
 ucs_status_t ucp_mem_reg_md_map_update(ucp_context_h context)
 {
     UCS_STRING_BUFFER_ONSTACK(strb, 256);
-    ucp_mem_dummy_handle_t memh = {
-        .memh.alloc_md_index = UCP_NULL_RESOURCE,
+    ucp_mem_dummy_handle_t reg_memh = {
+        .memh.alloc_md_index = UCP_NULL_RESOURCE
     };
+    ucp_mem_h alloc_memh;
     ucs_memory_type_t mem_type;
     ucp_md_index_t md_index;
     ucs_status_t status;
     uint8_t buff;
 
-    status = ucp_memh_register(context, &memh.memh,
+    status = ucp_memh_alloc(context, NULL, 1, UCS_MEMORY_TYPE_HOST,
+                            UCT_MD_MEM_ACCESS_ALL | UCT_MD_MEM_FLAG_HIDE_ERRORS,
+                            "get_alloc_md_id", &alloc_memh);
+    if (status != UCS_OK) {
+        return status;
+    }
+    context->alloc_md_index = alloc_memh->alloc_md_index;
```

Comment:
1. space line before
2. maybe the ucp_init function should first set it to some undefined value?